supabase/functions/on_channel_update/index.ts

```diff
@@ -27,8 +27,41 @@ serve(async (event: Request) => {
     const record = body.record
     console.log('record', record)
 
-    if (record.public) {
-      // find all other channels with same app_i with public true and update them to false
+    if (record.public && record.ios) {
+      const { error: iosError } = await supabaseAdmin()
+        .from('channels')
+        .update({ public: false })
+        .eq('app_id', record.app_id)
+        .eq('ios', true)
+        .neq('id', record.id)
+      const { error: hiddenError } = await supabaseAdmin()
+        .from('channels')
+        .update({ public: false })
+        .eq('app_id', record.app_id)
+        .eq('android', false)
+        .eq('ios', false)
+      if (iosError || hiddenError)
+        console.log('error', iosError || hiddenError)
+    }
+
+    if (record.public && record.android) {
+      const { error: androidError } = await supabaseAdmin()
+        .from('channels')
+        .update({ public: false })
+        .eq('app_id', record.app_id)
+        .eq('android', true)
+        .neq('id', record.id)
+      const { error: hiddenError } = await supabaseAdmin()
+        .from('channels')
+        .update({ public: false })
+        .eq('app_id', record.app_id)
+        .eq('android', false)
+        .eq('ios', false)
```

Comment:
We could re-use a base query rather than repeat it multiple times.

It'd be awesome do something as in the example below, to avoid repeat the query so many times

```suggestion
    const publicChannelsQuery = supabaseAdmin()
        .from('channels')
        .update({ public: false })
        .eq('app_id', record.app_id)
    if (record.public && record.ios) {
      const { error: iosError } = await publicChannelsQuery
        .eq('ios', true)
        .neq('id', record.id)
      const { error: hiddenError } = await publicChannelsQuery
        .eq('android', false)
        .eq('ios', false)
      if (iosError || hiddenError)
        console.log('error', iosError || hiddenError)
    }

    if (record.public && record.android) {
      const { error: androidError } = await publicChannelsQuery
        .eq('android', true)
        .neq('id', record.id)
      const { error: hiddenError } = await publicChannelsQuery
        .eq('android', false)
        .eq('ios', false)
```

Reply:
This will only work if you clone the query, you merged 2 queries into 1 and I think this is going to break the logic (?)

Reply:
Yeah, I'm not sure about the snippet but it'd be nice to have less duplicity on this code